src/Datadog.Trace.ClrProfiler.Managed/Emit/DynamicMethodBuilder.cs

```diff
@@ -197,6 +343,24 @@ internal static class DynamicMethodBuilder<TDelegate>
             return dynamicMethod.CreateDelegate();
         }
 
+        private static bool TypeArraysMatch(Type[] expectedTypes, int actualCount, Type[] actualTypes)
+        {
+            if (expectedTypes.Length != actualCount)
+            {
+                return false;
+            }
+
+            for (var i = 0; i < expectedTypes.Length; i++)
+            {
+                if (expectedTypes[i].AssemblyQualifiedName != actualTypes[i].AssemblyQualifiedName)
```

Comment:
Same comment here. I expect comparisons on the type objects themselves (using the Equals method) to be sufficient. Just a very minor "nitpick"